plugin/pkg/admission/taint/admission.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	if configmap == nil || configmap.Data["taint"] != "true" {
+		return nil
+	}
+
+	namespace := a.GetNamespace()
+	user := a.GetUserInfo()
+	tpmAdmin := false
+
+	if user == nil {
+		// Allow access over the insecure socket to do anything
+		tpmAdmin = true
+	} else {
+		// Otherwise check whether the user has a role that provides the tpmadmin attribute
+		rbac := newRbacClient(t.client)
+		ctx := api.WithUser(api.WithNamespace(api.NewContext(), "default"), user)
+		rules, err := rbac.GetEffectivePolicyRules(ctx)
+		if err == nil {
+			for _, rule := range rules {
+				restrictions, ok := rule.AttributeRestrictions.(*runtime.Unknown)
+				if ok && string(restrictions.Raw) == "tpmadmin" {
+					tpmAdmin = true
+					break
+				}
+			}
+		} else {
+			glog.Errorf("Unable to obtain user rules: %v", err)
+		}
+	}
+
+	// Allow admin users or request over the insecure socket to do whatever they want
+	if tpmAdmin == true {
+		glog.Errorf("TPMadmin = true")
+		return nil
+	}
+
+	// Unprivileged users aren't allowed to perform actions that would alter the trust state
+	switch kind {
+	case api.Kind("ConfigMap"):
+		name := a.GetName()
+		// A create operation may not provide a name directly, so pull it out of the object if we didn't get one
+		if name == "" {
+			configmap, ok := a.GetObject().(*api.ConfigMap)
+			if !ok {
+				return apierrors.NewBadRequest("Resource was marked with kind ConfigMap but couldn't be type converted")
+			}
+			name = configmap.Name
+		}
+		// Only permit TPM admins to modify these, otherwise nodes can circumvent security policy
+		if name == "taint.coreos.com" || name == "tpm.coreos.com" {
+				return apierrors.NewBadRequest("Unauthorised attempt to modify taint configuration")
+		}
+		return nil
+	case api.Kind("Node"):
+		operation := a.GetOperation()
+		if operation != admission.Create && operation != admission.Update {
+			return nil
+		}
+
+		node, ok := a.GetObject().(*api.Node)
+		if !ok {
+			return apierrors.NewBadRequest("Resource was marked with kind Node but couldn't be type converted")
+		}
+
+		if operation == admission.Create {
+			err := invalidateNode(node)
+			if err != nil {
+				return fmt.Errorf("Unable to invalidate node: %v", err)
+			}
+			return nil
+		}
+
+		// If an external update tries to switch a node from untrusted to trusted, force it back to untrusted
+		untrusted, err := isUntrusted(node)
+		if err != nil {
+			return fmt.Errorf("Unable to identify node trusted status: %v", err)
+		}
+		if untrusted == false {
+			oldNode, err := t.client.Core().Nodes().Get(node.Name)
+			if err != nil {
+				return fmt.Errorf("Attempting to update a node that doesn't exist? %v", err)
+			}
+			oldUntrusted, err := isUntrusted(oldNode)
+			if err != nil || oldUntrusted {
+				glog.Errorf("User %v attempted to flag untrusted node %v as trusted", user, node.Name)
+				invalidateNode(node)
+			}
+		}
+	case api.Kind("Pod"):
+		operation := a.GetOperation()
+		if operation != admission.Update {
```

Comment:
Also I don't know if the daemonset controller is taint aware -- in which case you could get daemonset pods assigned to untrusted nodes (another reason to potentially check on create as well)